ui/de-webapp/src/main/java/org/iplantc/de/server/controllers/DeController.java

```diff
@@ -46,12 +49,30 @@ public String redirectToDe(final HttpServletRequest request) throws MalformedURL
         return "redirect:de/";
     }
 
+    private DiscoveryEnvironmentMaintenance getDeMaintenance() {
+        return new DiscoveryEnvironmentMaintenance(maintenanceFile);
+    }
+
+    private boolean isLocalRequest(HttpServletRequest req) {
+        IpRanges ipRanges = new IpRanges(localIpRanges);
+        String forwardedFor = req.getHeader("X-Forwarded-For");
+        if (forwardedFor != null && forwardedFor.length() != 0) {
+            return ipRanges.matches(forwardedFor.split("\\s*,\\s*")[0]);
+        } else {
+            return ipRanges.matches(req.getRemoteAddr());
+        }
+    }
```

Comment:
Yeah, I could probably create a static function somewhere that takes a request and a set of IP ranges and checks them.